src/main/java/frc/utils/pathplannerutils/PathPlannerUtils.java

```diff
@@ -0,0 +1,114 @@
+import frc.utils.mirrorutils.MirrorableRotation2d;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class PathPlannerUtils {
+
+    public static void startPathPlanner() {
+        makePathPlannerCompatibleWithAdvantageKit();
+        scheduleWarmup();
+    }
+
+    private static void makePathPlannerCompatibleWithAdvantageKit() {
+        Pathfinding.setPathfinder(new LocalADStarAK());
+    }
+
+    private static void scheduleWarmup() {
+        PathfindingCommand.warmupCommand().schedule();
+    }
+
+    public static void configurePathPlanner(
+            Supplier<Pose2d> poseSupplier,
+            Consumer<Pose2d> resetPose,
+            Supplier<ChassisSpeeds> robotRelativeSpeedsSupplier,
+            Consumer<ChassisSpeeds> robotRelativeOutput,
+            HolonomicPathFollowerConfig config,
+            BooleanSupplier shouldFlipPath,
+            Subsystem driveSubsystem
+    ) {
+        AutoBuilder.configureHolonomic(
+                poseSupplier,
+                resetPose,//todo - trigon cancel
+                robotRelativeSpeedsSupplier,
+                robotRelativeOutput,
+                config,
+                shouldFlipPath,
+                driveSubsystem
+        );
+    }
+
+    public static void registerCommand(String commandName, Command command) {
+        NamedCommands.registerCommand(commandName, command);
+    }
+
+    public static void setDynamicObstacles(
+            List<Pair<MirrorablePose2d, MirrorablePose2d>> obstacles,
+            Pose2d currentRobotPose
+    ) {
+        List<Pair<Translation2d, Translation2d>> translationObstacles = obstacles.stream().map(pair ->
+                new Pair<>(
+                        pair.getFirst().get().getTranslation(),
+                        pair.getSecond().get().getTranslation()
+                )
+        ).collect(Collectors.toList());
+        Pathfinding.setDynamicObstacles(translationObstacles, currentRobotPose.getTranslation());
+    }
+
+    public static void setDynamicObstacles(Pair<MirrorablePose2d, MirrorablePose2d> obstacle, Pose2d currentRobotPose) {
+        setDynamicObstacles(Collections.singletonList(obstacle), currentRobotPose);
+    }
+
+    public static void removeAllDynamicObstacles(Pose2d currentRobotPose) {
+        List<Pair<MirrorablePose2d, MirrorablePose2d>> emptyList = new ArrayList<>();
+        setDynamicObstacles(emptyList, currentRobotPose);
+    }
+
+    public static void setTargetRotationOverride(Supplier<Optional<MirrorableRotation2d>> overrider) {
+        PPHolonomicDriveController.setRotationTargetOverride(() -> overrider.get().map(MirrorableRotation2d::get));
+    }
+
+    public static Command createOnTheFlyPathCommand(MirrorablePose2d currentPose, MirrorablePose2d targetPose,
```

Comment:
The current pose shouldn't be a mirrorable, it should always be relative to the blue alliance. Only the targetPose is mirrorable to account for object that are mirrored to the red side of the field.